### PR TITLE
Feature/hydra helm

### PR DESCRIPTION
**Pull request by Knorrke**
Introduce hydra using helm provider
this depends on serlo/infrastructure-modules-serlo.org#2 and serlo/infrastructure-modules-shared#2, which are atm referenced locally in the source

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -407,8 +421,94 @@ resource "kubernetes_namespace" "kpi_namespace" {
...
+
+  # ingress.annotations	Annotations for the ingress	{}
+  # ingress.path	Path of the ingress	/
+  # ingress.tls
+}
+
+data "helm_repository" "ory" {
+  name = "ory"
+  url  = "https://k8s.ory.sh/helm/charts"
+}
```
Comment (inyono) [inline main.tf]: I suppose that you still need to include the hydra module here?

**Comment by inyono** [comment]
I merged the required PRs and updated the references to the local modules to their respective github link.

File: main.tf (hunk 1 above)
Comment (Knorrke) [reply main.tf]: oh, I forgot to push my last commit^^

File: main.tf [hunk 2]
```diff
@@ -326,10 +340,18 @@ module "cloudflare" {
   }
 }
 
+module "hydra" {
+  source      = "github.com/serlo/infrastructure-modules-shared.git//hydra?ref=master"
+  dsn         = "postgres://${module.kpi.kpi_database_username_default}:${var.kpi_kpi_database_password_default}@${module.gcloud_postgres.database_private_ip_address}/hydra"
+  url_login   = "https://de.${local.domain}/auth/hydra/login"
+  url_consent = "https://de.${local.domain}/auth/hydra/consent"
+  namespace   = kubernetes_namespace.community_namespace.metadata.0.name
+  salt        = "1234567890123456789"
```
Comment (inyono) [inline main.tf]: This should probably be a random value or come from the secrets file, right?

**Review by inyono — COMMENTED** [review]
Is this working and ready for review now? :)

File: main.tf (hunk 2 above)
Comment (Knorrke) [reply main.tf]: I don't really remember what this is used for, only that it should never change https://www.ory.sh/docs/hydra/advanced#subject-identifier-algorithms. I guess moving it to the secrets file does make sense..?

**Comment by Knorrke** [comment]
Added the ingress config now, so hydra is available. This is ready for review now :-)

**Review by inyono — APPROVED** [review]

**Pull request merged**